### PR TITLE
Skeleton proposal

### PR DESCRIPTION
**Pull request by bubanoid**

### CODE REVIEW

**Comment by olnkravchenko** [comment]
@bubanoid @bulygin1985 can we move all the code into `src` folder and datasets into `tmp` folder instead of `VOC`?

File: training/train.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+dataset_val = torchvision.datasets.VOCDetection(
+    root="../VOC", year="2007", image_set="val", download=False
+)
+dataset_val = torchvision.datasets.wrap_dataset_for_transforms_v2(dataset_val)
+
+trainingdata_indices = torch.tensor(
+    [955, 1025, 219, 66, 1344, 222, 865, 2317, 86, 1409]
+)
+
+device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
```
Comment (bulygin1985) [inline training/train.py:19]: Please, use 2 modes:
1) overfit
2) train
Example:
import argparse
parser = argparse.ArgumentParser(description="Train parser")
parser.add_argument('o', '--overfit', action='store_true')

Each mode has own learning rate schedule

Comment (bubanoid) [reply training/train.py]: Done

File: training/train.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+model = ModelBuilder(alpha=0.25).to(device)
+
+parameters = list(model.parameters())
+optimizer = torch.optim.Adam(parameters, lr=lr)
+
+model.train(True)
+
+batch_generator = torch.utils.data.DataLoader(
```
Comment (bulygin1985) [inline training/train.py:57]: batch_size depends on parameter "overfit".

File: training/train.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    training_data, num_workers=4, batch_size=32, shuffle=True
```
Comment (bulygin1985) [inline training/train.py:58]: Please, check if overfit is reached if num_workers=4 .

File: training/train.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+)
+
+encoder = CenternetEncoder(input_height, input_width)
+
+dataset_val = torchvision.datasets.wrap_dataset_for_transforms_v2(dataset_val)
+torch_dataset = Dataset(dataset=dataset_val, transformation=transform, encoder=encoder)
+
+training_data = torch_dataset
+lr = 0.001
+desired_loss = 0.1
```
Comment (bulygin1985) [inline training/train.py:37]: For train mode it is impossible value. So if overfit =False, then train will be infinite. It is better to set learning rate schedule for overfit and for train modes

**Review by bulygin1985 — CHANGES_REQUESTED** [review]

File: training/train.py (hunk 2 above)
Comment (bubanoid) [reply training/train.py:57]: Done

File: training/train.py (hunk 3 above)
Comment (bubanoid) [reply training/train.py:58]: I'm not entirely sure I fully understand your suggestion. If you're asking whether setting num_workers=4 causes the program to crash, the answer is no — it works as expected.

File: training/train.py (hunk 4 above)
Comment (bubanoid) [reply training/train.py:37]: Done

Comment (bulygin1985) [reply training/train.py:37]: For train mode we have desired_loss = 0.1 but it is an impossible requirement. Please, set max_epoch as the criterion to finish train mode

**Review by bulygin1985 — CHANGES_REQUESTED** [review]

File: training/train.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+torch_dataset = Dataset(dataset=dataset_val, transformation=transform, encoder=encoder)
+
+training_data = torch_dataset
+lr = 0.03
+desired_loss = 0.1
+batch_size = 32
+
+if overfit:
+    # 10 randomly chosen pictures for model training
+    trainingdata_indices = torch.randperm(len(torch_dataset))[:10]
```
Comment (bulygin1985) [inline training/train.py:42]: it is better to fix the first 10 images because we need to visualize the result. However, the result could be verified only on the same images on which the model is overfitted.

**Review by bulygin1985 — CHANGES_REQUESTED** [review]

**Comment by macoidk** [comment]
Advice to fix

file: train.py
line: 41-43
Change the sample of 10 random pictures to the first 10 pictures

(p.s choosing random images complicates visualization, because then I have to look for which images the model was trained on)

File: training/train.py (hunk 4 above)
Comment (bubanoid) [reply training/train.py]: Fixed

File: training/train.py (hunk 5 above)
Comment (bubanoid) [reply training/train.py]: Changed training dataset to the first 10 images for the overfit case.

**Review by bulygin1985 — APPROVED** [review]
